### PR TITLE
End Game Functionality : WIP

### PR DESCRIPTION
**Pull request by glauroqj**
Please @pepper-seattle look that purpose to the end game/won game, it's not complete yet, check if you have some ideas to help me to finish this task, ok?

### CODE REVIEW

**Comment by pepper-seattle** [comment]
Looking now

**Comment by pepper-seattle** [comment]
Dude. This is awesome.

**Comment by pepper-seattle** [comment]
A few notes:
- If you click quickly through incorrect guesses the animation can't keep up
- The winning end game is cool, think we can link to wikipedia or monster wiki articles instead of a google search?
- I think the alerts work as an initial implementation for win/loss, but let's aim to change those out with a modal or toast in another PR (not here, this is already a lot)
- Instead of the text 'chances remaining' consider centering the number, making it larger, and having the skull overlay it as it drops with incorrect guesses or play around with where else it could show up
-- There are also wide bands above and/or below the image you could toy with having a timer and the remaining guesses appear there

File: src/App.js [hunk 1]
```diff
@@ -12,23 +12,85 @@ import NewGameButton from './components/NewGameButton/NewGameButton'
...
+    dispatch( resetGame() )
+    startGame()
+  }
+
+  const gameWinAction = () => {
+    alert('< YOU WOON, GET MY PERK!! >')
+
+    const wordToSearch = displayValues.reduce((acc, cur) => acc + cur, '')
+
+    window.open(`https://www.google.com/search?q=${wordToSearch} cthulhu`)
```
Comment (pepper-seattle) [inline src/App.js:80]: Brilliant, but let's hone this down to wikipedia or monster wiki or something we know will have all the definitions available

**Comment by glauroqj** [comment]
hello @pepper-seattle I make some changes, see bellow:

- change the effect when decrease chances
- remove text: chances remaining
- when decrease chances now is overlayed off remaining numbers

Well, I liked a lot about the idea off wikipedia or monster wiki, could you help me with some links? Because idk if wikipedia has all the monsters listed in the game.

About the alert and timing, I would like to create another PR with this features.

File: src/App.js (hunk 1 above)
Comment (glauroqj) [reply src/App.js]: Well, could you help me with some example links? I really dont know if wikipedia covers all the monsters in game.

**Comment by glauroqj** [comment]
Thinking in some further features, I would like to create a ranking for logged players, what do you think?

File: src/App.js (hunk 1 above)
Comment (pepper-seattle) [reply src/App.js]: Did a bit of testing on this one and without some refactoring it'll be hard to direct link these outside of google. Sorry bout that, you can change the linking back to google or leave it as is and I'll fix it

**Comment by pepper-seattle** [comment]
Hi @glauroqj,
The changes are good. As I responded above we'll have to go back to the google linking unfortunately. I have an idea, but it'll take a decent refactor of how we consume the data at linkage to do it right. So no need to worry about that here, just bring back the google linking.
I'm getting a bit of doubling on the animation where it'll appear as expected and then right after partially appear in a spot above it's original location, but still over the number. Let's allow it to hang on a little bit longer and see if that fixes it.
Sorry for the delays in review, still working a full time job and it's been crazy lately. Will be better about responding in the same day at least from here on out.

**Comment by glauroqj** [comment]
Hello @pepper-seattle , no problem about the delay man, it's completely understood.

Well, about the animation, I changed it for .8s take a look please.

Do you need any other changes before merge this?

File: src/App.js [hunk 2]
```diff
@@ -12,23 +12,84 @@ import NewGameButton from './components/NewGameButton/NewGameButton'
...
+  const [chances, setChances] = useState(0)
+  const [removeChance, setRemoveChance] = useState(false)
+
+  useEffect(() => {
+    startGame()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
 
   useEffect(() => {
+    // console.log('< COUNT > ', guessCount, guessedLetters)
```
Comment (pepper-seattle) [inline src/App.js:32]: Make sure to get rid of your various console logs

File: src/App.js [hunk 3]
```diff
@@ -12,23 +12,84 @@ import NewGameButton from './components/NewGameButton/NewGameButton'
...
+    return array1 === array2 ? gameWinAction() : false
+  }
+
+
+  const startGame = () => {
     const wordToGuess = selectWordToGuess(data, window.innerWidth)
     dispatch( setWordToGuess(wordToGuess) )
-  }, [dispatch])
+    /** create chances */
+    setChances((Number(wordToGuess.length) * 1.5 < 26 ) ? Math.floor(wordToGuess.length * 1.5)  : 23)
```
Comment (pepper-seattle) [inline src/App.js:65]: nice

File: src/App.js [hunk 4]
```diff
@@ -12,23 +12,84 @@ import NewGameButton from './components/NewGameButton/NewGameButton'
...
+  }
+
+  const gameOverAction = () => {
```
Comment (pepper-seattle) [inline src/App.js:68]: Should have actions living in the actions file

File: src/state/constants.js [hunk 5]
```diff
@@ -2,4 +2,5 @@ export const GUESS_COUNT_INCREASE = "GUESS_COUNT_INCREASE";
 export const GUESS_LETTER = "GUESS_LETTER";
 export const SET_DISPLAY_VALUES = "SET_DISPLAY_VALUES";
 export const SET_WORD_TO_GUESS = "SET_WORD_TO_GUESS";
-export const START_GAME = "START_GAME";
\ No newline at end of file
+export const START_GAME = "START_GAME";
+export const RESET_GAME = "RESTE_GAME"
```
Comment (pepper-seattle) [inline src/state/constants.js:6]: misspelled "reset" here as "RESTE"

**Comment by pepper-seattle** [comment]
Left a few comments and if you fix those then I will merge. Sorry again for the delay!

**Comment by glauroqj** [comment]
Hello @pepper-seattle no problem. I did some changes as you ask, please check it out.

**Comment by pepper-seattle** [comment]
Good work, merging!

**Pull request merged**